src/dcap_provider.cpp

```diff
@@ -861,24 +1085,82 @@ static quote3_error_t convert_to_intel_error(sgx_plat_error_t platformError)
 
 static std::string build_pck_crl_url(
     std::string crl_name,
-    std::string api_version)
+    std::string api_version,
+    sgx_prod_type_t prod_type = SGX_PROD_TYPE_SGX)
 {
-    std::string version = get_collateral_version();
+    std::string version;
+    std::stringstream url;
+    std::string client_id;
+    if (prod_type == SGX_PROD_TYPE_TDX)
+    {
+        version = get_collateral_version_tdx();
+
+        client_id = get_client_id();
+
+        url << get_base_url_tdx();
+
+        if (!version.empty())
+        {
+            url << "/" << version;
+        }
+
+        url << "/pckcrl?ca=" << crl_name << "&";
+
+        if (!client_id.empty())
+        {
+            url << "clientid=" << client_id;
+        }
+	}
+    else
+    {
+        version = get_collateral_version();
+
+        std::string escaped =
+            curl_easy::escape(crl_name.data(), (int)crl_name.size());
+
+        client_id = get_client_id();
+
+        url << get_base_url();
+
+        if (!version.empty())
+        {
+            url << "/" << version;
+        }
+
+        url << "/pckcrl?uri=" << escaped << "&";
+
+        if (!client_id.empty())
+        {
+            url << "clientid=" << client_id << '&';
+        }
+
+        url << api_version;
+	}
+
+    return url.str();
+}
+
+static std::string build_root_ca_crl_url_tdx()
+{
+    std::string version = get_collateral_version_tdx();
+
     std::stringstream url;
-    std::string escaped =
-        curl_easy::escape(crl_name.data(), (int)crl_name.size());
     std::string client_id = get_client_id();
-    url << get_base_url();
+
+    url << get_base_url_tdx();
+

```

Comment:
same as above, if version is empty, throw "SGX_QL_ERROR_INVALID_PARAMETER" else append the version

Reply:
Removing the check for the reasons listed above.